### PR TITLE
Change GPUError to a base class instead of a union

### PR DESCRIPTION
**Pull request by kainino0x**
Closes #2750
Related to #1884

***
<a href="https://pr-preview.s3.amazonaws.com/kainino0x/gpuweb/pull/2853.html" title="Last updated on May 10, 2022, 11:45 PM UTC (19ddaeb)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/2853/a81c1d8...kainino0x:19ddaeb.html" title="Last updated on May 10, 2022, 11:45 PM UTC (19ddaeb)">Diff</a>

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -10252,20 +10252,44 @@ enum GPUErrorFilter {
...
 
 <script type=idl>
+[Exposed=(Window, DedicatedWorker), SecureContext]
+interface GPUError {
+    constructor(DOMString message);
+    readonly attribute DOMString message;
+};
+
 [Exposed=(Window, DedicatedWorker), SecureContext]
 interface GPUOutOfMemoryError {
```
Comment (toji) [inline spec/index.bs:10262]:
```suggestion
interface GPUOutOfMemoryError : GPUError {
```

Isn't this the intent?

File: spec/index.bs [hunk 2]
```diff
@@ -10252,20 +10252,44 @@ enum GPUErrorFilter {
...
-    constructor();
 };
 
 [Exposed=(Window, DedicatedWorker), SecureContext]
 interface GPUValidationError {
```
Comment (toji) [inline spec/index.bs:10266]:
```suggestion
interface GPUValidationError : GPUError {
```

File: spec/index.bs (hunk 1 above)
Comment (kainino0x) [reply spec/index.bs:10262]: Oops, yes, thanks.

**Review by toji — APPROVED** [review]
LGTM!

**Review by kdashg — APPROVED** [review]

**Comment by lokokung** [comment]
Updated CTS draft issue `val: error scope`, so removing 'needs-cts-label'.

**Pull request merged**